File: LeetCodeProblemsGo/LeetCode2169CountOperationsToObtainZero/LeetCode2169CountOperationsToObtainZero.go
package LeetCode2169CountOperationsToObtainZero

func countOperations(num1 int, num2 int) int {
	counter := 0
	for num1 != 0 && num2 != 0 {
		if num1 >= num2 {
			num1 -= num2
		} else {
			num2 -= num1
		}
		counter++
	}
	return counter
}

File: LeetCodeProblemsCMake/Problems/LeetCode2169CountOperationsToObtainZero.cpp
//
// Created by saeed on 11/9/25.
//

#include "LeetCode2169CountOperationsToObtainZero.h"

int LeetCode2169CountOperationsToObtainZero::countOperations(int num1, int num2) {
    int counter{};
    while (num1 != 0 && num2 != 0) {
        if (num1 >= num2) {
            num1 -= num2;
        } else {
            num2 -= num1;
        }
        ++counter;
    }
    return counter;
}
